client/client.go

```diff
@@ -674,7 +674,7 @@ func (r *NotaryRepository) publish(cl changelist.Changelist) error {
 	if snapshotJSON, err := serializeCanonicalRole(
 		r.tufRepo, data.CanonicalSnapshotRole, nil); err == nil {
 		// Only update the snapshot if we've successfully signed it.
-		updatedFiles[data.CanonicalSnapshotRole] = snapshotJSON
```

Comment:
Non-blocking:  this is totally fine, but it might be easier to change `updatedFiles` to be a `map[data.RoleName][]byte`, so below we don't have to call `data.CanonicalRootRole.String()`.  This would also mean that `signTargets` wouldn't have to cast back to `data.RoleName` before calling `serializeCanonicalRole`.